packages/ui-kit/src/components/mermaid-client-side.tsx

```diff
@@ -0,0 +1,134 @@
+    textColor: colors.light.textPrimary,
+    fontFamily: typography.fontFamilies.primary,
+    fontSize: typography.fontSizes.body,
+    primaryColor: customProperties.colorInfo95,
+    primaryBorderColor: customProperties.colorInfo,
+    primaryTextColor: colors.light.textPrimary,
+
+    secondaryColor: customProperties.colorAccent40,
+    secondaryBorderColor: customProperties.colorAccent30,
+    secondaryTextColor: colors.light.textPrimary,
+
+    tertiaryColor: customProperties.colorPrimary95,
+    tertiaryBorderColor: customProperties.colorPrimary25,
+    tertiaryTextColor: colors.light.textPrimary,
+
+    // notes are conventionally yellow but there is none in the design system
+    noteBkgColor: 'lightyellow',
+    noteTextColor: colors.light.textPrimary,
+    noteBorderColor: 'yellow',
+
+    errorBkgColor: colors.light.surfaceError,
+    errorTextColor: colors.light.textError,
+
+    // sequence diagram specifics:
+    // https://mermaid-js.github.io/mermaid/#/theming?id=sequence-diagram
+    sequenceNumberColor: colors.light.textInverted,
+    actorLineColor: colors.light.borderInfo,
+
+    // flow chart specifics:
+    // https://mermaid-js.github.io/mermaid/#/theming?id=flowchart
+    edgeLabelBackground: colors.light.surfaceSecondary1,
+  },
+  flowchart: {
+    useMaxWidth: true,
+    htmlLabels: true,
+    curve: 'cardinal',
+  },
+  sequence: {
+    // https://mermaid-js.github.io/mermaid/#/sequenceDiagram?id=possible-configuration-parameters
+    fontFamily: typography.fontFamilies.primary,
+    mirrorActors: false,
+    useMaxWidth: true,
+    rightAngles: false,
+    showSequenceNumbers: true,
+    diagramMarginX: 5, // prevent border clipping
+    diagramMarginY: 5, // prevent border clippings
+    bottomMarginAdj: 2,
+    boxTextMargin: 5,
+    noteMargin: 10,
+    messageMargin: 35,
+
+    actorFontSize: typography.fontSizes.body,
+    actorFontWeight: typography.fontWeights.regular,
+    messageFontSize: typography.fontSizes.body,
+    messageFontFamily: typography.fontFamilies.primary,
+    messageFontWeight: typography.fontWeights.regular,
+    noteFontSize: typography.fontSizes.small,
+  },
+  gantt: {
+    numberSectionStyles: 4,
+    axisFormat: '%Y-%m-%d',
+    fontFamily: typography.fontFamilies.primary,
+  },
+  // this low level CSS override is rather a bugfix workaround, resetting fill and stroke to fix font rendering.
+  themeCSS: `
+  .messageText {
+    color: ${colors.light.textSecondary}
+    fill: inherit;
+    stroke: inherit;
+  }
+`,
+} as const;
+
+const Figure = styled.figure`
+  background-color: ${colors.light.surfaceSecondary1};
+  border-radius: ${tokens.borderRadiusForImageFrame};
+  margin: 0;
+  padding: ${dimensions.spacings.xs};
+  display: flex;
+  justify-content: center;
+  line-height: normal;
+  a span.nodeLabel {
+    color: ${colors.light.link} !important;
+    text-decoration: underline;
+  }
+`;
+
+const idForGraph = (graph: string) => `mermaid-${murmurhash.v3(graph)}`;
```

Comment:
murmurhash is a compact hash implementation that is optimized for being collision free.  The package is smaller than the wrappers around nodejs crypto APIs that could have provided md5 or sha and it's the right hash for the use case (mermaid needs a unique ID for every graph to not get confused)